src/ota/interface/OTAInterfaceDefault.cpp

```diff
@@ -94,19 +93,26 @@ OTACloudProcessInterface::State OTADefaultCloudProcessInterface::fetch() {
   uint32_t start = millis();
 
   do {
-    if(http_client->available() == 0) {
-      goto exit;
-    }
+    if(http_client->available()) {
+      http_res = http_client->read(context->buffer, context->buf_len);
+
+      if(http_res < 0) {
+        DEBUG_VERBOSE("OTA ERROR: Download read error %d", http_res);
+        res = OtaDownloadFail;
+        goto exit;
+      }
```

Comment:
```suggestion
    if(http_client->available() == 0) {
      /* Avoid tight loop and allow yield */
      delay(1);
      continue;
    }

    http_res = http_client->read(context->buffer, context->buf_len);

    if(http_res < 0) {
      DEBUG_VERBOSE("OTA ERROR: Download read error %d", http_res);
      res = OtaDownloadFail;
      goto exit;
    }
```

Reply:
personally i think the `if else` solution is more intuitive, but i also think a `continue` will feel good in this `goto` function :smile: